client/src/modules/list/components/Items/ListItemName/ListItemName.jsx

```diff
@@ -0,0 +1,167 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { updateListItem } from '../model/actions';
+import { RouterMatchPropType } from 'common/constants/propTypes';
+import { KeyCodes } from 'common/constants/enums';
+import Preloader from 'common/components/Preloader';
+
+class ListItemName extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    const { name } = this.props;
+
+    this.state = {
+      isNameInputFocused: false,
+      isTipVisible: false,
+      name,
+      pending: false
+    };
+
+    this.nameInput = React.createRef();
+    this.listItemName = React.createRef();
+  }
+
+  componentDidUpdate() {
+    const { isNameInputFocused, name } = this.state;
+
+    if (isNameInputFocused) {
+      document.addEventListener('keydown', this.handleKeyPress);
+      document.addEventListener('mousedown', this.handleClickOutside);
+    }
+
+    if (!isNameInputFocused) {
+      document.removeEventListener('keydown', this.handleKeyPress);
+      document.removeEventListener('mousedown', this.handleClickOutside);
+    }
```

Comment:
these 'if' blocks above should go to: 'handleNameInputFocus' and 'handleNameInputBlur'